lib/codebreaker_ai/validators.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module CodebreakerAi
+  class AbstractValidator
+    def initialize
+      raise AbstractClassError, I18n.t(:"error.class.abstarct_class")
+    end
+
+    def valid(_value)
+      raise AbstractMethodError, I18n.t(:"error.class.abstarct_method")
+    end
+
+    def message
+      raise AbstractMethodError, I18n.t(:"error.class.abstarct_method")
+    end
+  end
+
+  class NameValidator < AbstractValidator
+    def initialize; end
+
+    def valid(name)
+      name.instance_of?(String) && name.length.between?(3, 20)
+    end
+
+    def message
+      I18n.t(:"validator.invalid_name")
+    end
+  end
+
+  class DifficultyValidator < AbstractValidator
+    def initialize; end
+
+    def valid(difficulty)
+      DIFFICULTIES.include? difficulty.to_sym.downcase
+    end
+
+    def message
+      I18n.t(:"validator.invalid_difficulty")
+    end
+  end
+
+  class ConfirmationValidator < AbstractValidator
+    CONFIRMATION = {
+      y: 'y',
+      yes: 'yes',
+      n: 'n',
+      no: 'no'
+    }.freeze
+
+    def initialize; end
+
+    def valid(value)
+      CONFIRMATION.include? value.to_sym.downcase
+    end
+
+    def message
+      format('%<message>s %<title>s: %<values>s',
+             message: I18n.t(:"validator.invalid_confirmation"),
+             title: I18n.t(:allowed_values),
+             values: self.CONFIRMATION.to_a.join(', '))
+    end
+  end
+
+  class GuessValidator < AbstractValidator
+    attr_reader :message
+
+    def initialize; end
+
+    def valid(value)
```

Comment:
u can just check on regex for 4-digits number in range of 1-6
all this data u have in constants